scalameta/io/js/src/main/scala/scala/meta/internal/io/PlatformIO.scala

```diff
@@ -4,10 +4,37 @@ import java.nio.charset.Charset
 import scala.meta.io._
 
 object PlatformIO {
-  def workingDirectory: AbsolutePath = AbsolutePath(fileSeparator).get
-  def slurp(path: AbsolutePath, charset: Charset): String = slurp(path)
+  def fileSeparator: String =
+    "/"
+
+  def pathSeparator: String =
+    ":"
+
+  def workingDirectory: AbsolutePath =
+    AbsolutePath("/")
+
+  def isAbsolutePath(path: String): Boolean =
+    path.startsWith("/")
+
+  def isRelativePath(path: String): Boolean =
+    !path.startsWith("/")
+
+  def absolutize(parent: AbsolutePath, path: RelativePath): AbsolutePath = {
+    AbsolutePath(parent.toString + fileSeparator + path.toString)
+  }
+
+  def relativize(parent: AbsolutePath, path: AbsolutePath): RelativePath = {
```

Comment:
Yes, there's quite some duplication between js/jvm code in PlatformIO. You should've seen me when I hacked that functionality at 3am. That's clearly unacceptable, and I'll look into that soon.